tensorboard/uploader/proto/write_service.proto

```diff
@@ -119,6 +124,125 @@ message WriteScalarResponse {
+//  * The caller writes each blob in the sequence separately via the WriteBlob
+//    RPC as described below, providing the blob sequence ID and sequence index
+//    for each blob.
+//  * Multiple WriteBlob streams may operate in parallel, with no constraint
+//    on their ordering.
+
+// Blob sequence uploads are resumable, as follows:
+//  * The client starts the upload process with a new CreateBlobSequence RPC.
+//  * The blob sequences are uniquely keyed by experiment, run, tag, and step.
+//    Since this is a second (or later) attempt regarding an existing sequence,
+//    a new sequence is not created.  Instead the existing sequence ID is
+//    returned.
+//  * The client attempts to upload each blob in the sequence.  These uploads
+//    resume or may be skipped entirely, as needed, according to the below
+//    procedure.
+
+// The blob writing procedure is as follows.  It is intrinsically resumable.
+//  * The caller instantiates a WriteBlob RPC, which is a bidirectional stream.
+//  * The first of these requests must provide write_blob_first_request_data,
+//    and the subsequent ones must not.
+//  * The first request should send no actual data; it acts as a request for the
+//    current status of the blob, in case it has already been partially or fully
+//    uploaded.
+//  * The first WriteBlobResponse reports the state, size, and crc32c of the
+//    object so far, i.e. taking into account any previously uploaded data.
+//  * The client proceeds to upload from the specified offset (optionally
+//    validating the crc32c up to that point first).
+//  * The caller sends a number of additional WriteBlobRequests, providing the
+//    blob data in an ordered sequence of chunks.
+//  * The caller receives a sequence of WriteBlobResponses in return.  Each
+//    WriteBlobResponse corresponds to one WriteBlobRequest in order, reporting
+//    the state of the write operation.
+//  * In general, the caller should not wait to verify that each response
+//    reports the expected status, size, and crc32c before sending the next
+//    request.  Doing so harms throughput, with little benefit since any
+//    mismatch on the server side will cause the RPC to throw an error anyway.
+//  * The request size should be less than 4 MiB (2^20 = 4,194,304), because
+//    that is the default request limit for gRPC. Accounting for the other
+//    fields of the request, then, and leaving some padding for safety, the data
+//    chunk should be somewhat smaller-- perhaps simply 4e6 bytes.
+//  * The last of these requests must set finalize_object=True, and the prior
+//    ones must not.
+//  * For a blob <= 4MB, a single data chunk suffices, so there are two
+//    requests: one to learn the status, and one to upload the data.
+//
+// Note that the WriteBlobRequest does not mirror the nested structure of
+// WriteScalarRequest, because we only ever send one blob at a time.
+
+message CreateBlobSequenceRequest {
+  string experiment_id = 1;
```

Comment:
Nit: could we get a comment for each field?  Can be a terse 1-liner but still worth having IMO per https://aip.dev/192

Reply:
Done.